fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/lock/LockedObject.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common.util.concurrent.lock;
+
+import java.util.List;
+
+public class LockedObject<T> extends AutoCloseableLock {
```

Comment:
Do not couple metadata operations with locking operations

Reply:
Since Lock is an independent module and does not rely on a specific database or table, you can encapsulate simple functions within a specific class. However, avoid introducing metadata-related content into the Lock package.